test/test_writer.py

```diff
@@ -616,7 +630,7 @@ def test_request_response_concur(self, is_gzip):
     def test_response_warc_1_1(self, is_gzip):
         writer = BufferWARCWriter(gzip=is_gzip, warc_version='WARC/1.1')
 
-        resp = sample_response(writer)
+        resp = sample_response(FixedTestRecordBuilder(warc_version='WARC/1.1'))
```

Comment:
Similarly, can have two tests here, one that does `sample_response(writer)` and one that does 
`sample_response(FixedTestRecordBuilder(warc_version='WARC/1.1'))` so that we can be sure the warc_version is passed from the writer class and also works with builder directly